lib/pure/asynchttpserver.nim

```diff
@@ -300,9 +300,10 @@ proc processRequest(
           break
 
         # Read bytesToRead and add to body
-        # Note we add +2 because the line must be terminated by \r\n
-        let chunk = await client.recv(bytesToRead + 2)
-        request.body = request.body & chunk
+        let chunk = await client.recv(bytesToRead)
+        request.body.add(chunk)
+        # Skip \r\n (chunk terminating bytes per spec)
```

Comment:
can't we at least `assert` that we got \r\n ?

Reply:
Sure, I'd be happy to. I'm not 100% familiar with Nim's asserts, but it seems like we'd need slightly different behaviour if they're being compiled out or not.

Thoughts on something like:
```
when not defined(assert):
  discard await client.recv(2)
else:
  let separator = await client.recv(2)
  assert(separator == "\r\n\")
```

Reply:
use `when compileOption("assertions")`

Reply:
Fixed.